implementations/rust/ockam/ockam_command/src/tcp/outlet/show.rs

```diff
@@ -94,13 +100,44 @@ impl ShowTui {
         opts: CommandGlobalOpts,
         cmd: ShowCommand,
     ) -> miette::Result<()> {
-        let node = BackgroundNodeClient::create(&ctx, &opts.state, &cmd.node_opts.at_node).await?;
+        let node = BackgroundNodeClient::create(&ctx, &opts.state, &cmd.at).await?;
+        let node_name = node.node_name();
+        let opts_clone = opts.clone();
+
         let tui = Self {
             ctx,
             opts,
             cmd,
             node,
         };
+
+        // Display a message with the node name, and early return, if no TCP Outlets are
+        // found.
+        let is_finished: Mutex<bool> = Mutex::new(false);
+        let future_outlet_names = async {
+            let it: Vec<String> = tui.list_items_names().await?;
+            *is_finished.lock().await = true;
+            Ok(it)
+        };
+        let output_messages = vec![format!(
+            "Listing TCP Outlets on node {}...\n",
+            color_primary(node_name.clone())
+        )];
+        let progress_output = opts_clone
+            .terminal
+            .progress_output(&output_messages, &is_finished);
+        let (outlets, _) = try_join!(future_outlet_names, progress_output)?;
+        if outlets.is_empty() {
+            tui.terminal()
+                .stdout()
+                .plain(fmt_info!(
+                    "No TCP Outlets found on node {}",
+                    color_primary(node_name.clone())
+                ))
+                .write_line()?;
+            return Ok(());
+        }
```

Comment:
This is pretty much what the `tui.show()` function does in its first lines. We use that `Tui` trait exactly to avoid repeating this in every show command.

Would it be OK to change the message [here](https://github.com/build-trust/ockam/blob/04cf73ab070197a6502e547da66e26adbe063494/implementations/rust/ockam/ockam_command/src/terminal/tui.rs#L26)?

Reply:
great idea! done